lib/gears/filesystem.lua

```diff
@@ -11,20 +11,30 @@ local gtable = require("gears.table")
 
 local filesystem = {}
 
+local function make_parent(gfile)
+    local success, err = gfile:make_directory_with_parents()
+    if success then
+        return true
+    end
+    if err.domain == Gio.IOErrorEnum and err.code == "EXISTS" then
+        -- Directory already exists, let this count as success
+        return true
+    end
+    return false, err
+end
+
 --- Create a directory
 -- @tparam string dir The directory.
 -- @return (true, nil) on success, (false, err) on failure
 function filesystem.mkdir(dir)
```

Comment:
What do you think about merging `mkdir` and `make_parent_dir` since they do essentially the same thing (making directories)? Adding an extra parameter to `mkdir` wouldn't break API.

Something like:
```lua
function filesystem.mkdir(path, parents_only)
  if parents_only then
    return make_parent(Gio.File.new_for_path(path):get_parent())
  end
  return make_parent(Gio.File.new_for_path(path))
end
```